TheraLang.IntegrationTests/ControllersTests/UserControllerTests.cs

```diff
@@ -0,0 +1,165 @@
+
+        [Fact]
+        public async Task GetAllUsers_FailureUnauthorized()
+        {
+            var request = _baseUrl;
+            var response = await UnauthorizedClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+        [Fact]
+        public async Task GetAllUsers_FailNotAdmin()
+        {
+            var request = _baseUrl;
+            var response = await MemberClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetAdminUserView_FailureUnauthorized()
+        {
+            var request = $"{_baseUrl}/admin/{DefaultValues.MemberId}";
+            var response = await UnauthorizedClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetAdminUserView_FailNotAdmin()
+        {
+            var request = $"{_baseUrl}/admin/{DefaultValues.MemberId}";
+            var response = await MemberClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetAdminUserView_SuccessStatusCode()
+        {
+            var request = $"{_baseUrl}/admin/{DefaultValues.MemberId}";
+            var response = await AdminClient.GetAsync(request);
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task GetRoles_SuccessStatusCode()
+        {
+            var request = $"{_baseUrl}/roles";
+            var response = await AdminClient.GetAsync(request);
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task GetRoles_FailureUnauthorized()
+        {
+            var request = $"{_baseUrl}/roles";
+            var response = await UnauthorizedClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetRoles_FailNotAdmin()
+        {
+            var request = $"{_baseUrl}/roles";
+            var response = await MemberClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetUserRole_SuccessStatusCode()
+        {
+            var request = $"{_baseUrl}/{DefaultValues.MemberId}/role";
+            var response = await AdminClient.GetAsync(request);
+            response.EnsureSuccessStatusCode();
+        }
+
+        [Fact]
+        public async Task GetUserRole_FailureUnauthorized()
+        {
+            var request = $"{_baseUrl}/{DefaultValues.MemberId}/role";
+            var response = await UnauthorizedClient.GetAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetUserRole_FailNotAdmin()
```

Comment:
You really need both FailNotAdmin Test and FailUnauthorized tests? I think FailNotAdmin is enough.

Reply:
Its different behavior, in future if someone will accidentlly take away this role = "Andmin" test will not work. If not it will.

Reply:
I understand, but if someone remove Roles = "Admin", but it's feature is available only for admin there is no sense in FailUnauthorized test, cause FailNotAdmin test will fail. I agree that FailNotAdmin is important test, but if you have this one you don't need FailUnauthorized.

Reply:
agree, okay